docs/core/versions/index.md

```diff
@@ -82,6 +82,16 @@ Preview versions have a `-preview.[number].[build]` appended to the version numb
 
 After a release goes out, the release branches generally stop producing daily builds and instead start producing servicing builds. Servicing versions have a `-servicing-[number]` appended to the version. For example, `5.0.1-servicing-006924`.
 
+## .NET Runtime Compatibility
+
+.NET runtime maintains a high level of compatibility between versions. .NET applications should by and large, continue to work after upgrade to a new major .NET runtime version.
+
+Each major .NET runtime version contains intentional, carefully vetted and documented [breaking changes](https://learn.microsoft.com/en-us/dotnet/core/compatibility/breaking-changes). The documented breaking changes are not the only potential source of issues affecting the application after upgrade. For example, a performance improvement in .NET runtime (that is not considered to be a breaking change) can expose latent application threading bugs resulting in an application no longer working on a new .NET runtime major version. It is expected for very large applications to require a few fixes after upgrade to a new .NET runtime major version.
+
+The .NET runtime applications are configured to run on a given .NET runtime major version by default and recompilation is required to upgrade the application to run on a new .NET runtime major version. It is strongly recommended to re-test the application after upgrade to identify issues affecting the application.
+
+If upgrading via app recompilation is not feasible, the .NET runtime provides [additional settings](selection.md#control-roll-forward-behavior) to enable application to run on a higher major .NET runtime version than the version application was compiled for. These setting do not change the risks involved in upgrading the application to a higher major .NET runtime version and it is still strongly recommended to retest the application after upgrade.
+
```

Comment:
This section does not differentiate between dependent assemblies from applications.  I feel this guidance needs the nuance to accommodate the whole application.  A case study is .NET standard libraries.  Do they require rebuild to target a different major version (the assembly never had a major version designation)?  I would say no.  Taking this one step further, the .NET 7 app that uses this library needs to be validated on the major version it will be deployed, but all the dependencies do not require recompilation.  Thus going back to if the primary assembly that has the static entry point would also require recompilation.

Reply:
I have added note about libraries. Could you please take a look?

Reply:
That was a good addition.